test/core.jl

```diff
@@ -2239,10 +2240,10 @@ f9534d() = (a=(1,2,4,6,7); a[7])
 f9534d(x) = (a=(1,2,4,6,7); a[x])
 @test try; f9534d() catch ex; (ex::BoundsError).a === (1,2,4,6,7) && ex.i == 7; end
 @test try; f9534d(-1) catch ex; (ex::BoundsError).a === (1,2,4,6,7) && ex.i == -1; end
-f9534e(x) = (a=IOBuffer(); a.(x) = 3)
+f9534e(x) = (a=IOBuffer(); setfield!(a, x, 3))
 @test try; f9534e(-2) catch ex; is((ex::BoundsError).a,Base.IOBuffer) && ex.i == -2; end
```

Comment:
the BoundsError is from attempting to access the properties of field `-2` of the IOBuffer type